resources/default.rb

```diff
@@ -1,18 +1,125 @@
-actions :install, :uninstall
-default_action :install
-
-attribute :servicename, kind_of: String, name_attribute: true
-attribute :user, kind_of: [String, NilClass]
-attribute :group, kind_of: [String, NilClass]
-attribute :password, kind_of: [String, NilClass], sensitive: true
-attribute :url, kind_of: String, default: lazy { node['nexus3']['url'] }
-attribute :checksum, kind_of: String, default: lazy { node['nexus3']['checksum'] }
-attribute :path, kind_of: String, default: lazy { node['nexus3']['path'] }
-attribute :data, kind_of: String, default: lazy { node['nexus3']['data'] }
-attribute :home, kind_of: String, default: lazy { node['nexus3']['home'] }
-attribute :properties_cookbook, kind_of: String, default: lazy { node['nexus3']['properties_cookbook'] }
-attribute :properties_source, kind_of: String, default: lazy { node['nexus3']['properties_source'] }
-attribute :properties_variables, kind_of: Hash, default: lazy { node['nexus3']['properties_variables'] }
-attribute :vmoptions_cookbook, kind_of: String, default: lazy { node['nexus3']['vmoptions_cookbook'] }
-attribute :vmoptions_source, kind_of: String, default: lazy { node['nexus3']['vmoptions_source'] }
-attribute :vmoptions_variables, kind_of: Hash, default: lazy { node['nexus3']['vmoptions_variables'] }
+property :instance_name, kind_of: String, name_attribute: true
+property :nexus3_user, kind_of: [String, NilClass], default: lazy { node['nexus3']['user'] }
+property :nexus3_group, kind_of: [String, NilClass], default: lazy { node['nexus3']['group'] }
+# property :nexus3_password, kind_of: [String, NilClass], sensitive: true # Admin password
+property :version, kind_of: String, default: lazy { node['nexus3']['version'] }
+property :url, kind_of: String, default: lazy { node['nexus3']['url'] }
+property :checksum, kind_of: String, default: lazy { node['nexus3']['checksum'] }
+property :nexus3_home, kind_of: String, default: lazy { node['nexus3']['home'] }
+property :path, kind_of: String, default: lazy { node['nexus3']['path'] }
+property :data, kind_of: String, default: lazy { node['nexus3']['data'] }
+# Name of service used as property for nexus3_service; can be instance_name or something different
+# like version number.
+property :service_name, kind_of: String, default: lazy { instance_name }
```

Comment:
I am not sure I understand.

If you suggest writing `property :service_name, kind_of: String, default: lazy { new_resource.instance_name }`, this is not working: ``nexus3[foo] (nexus3_test::default line 6) had an error: NoMethodError: undefined method `new_resource' for Custom resource nexus3 from cookbook nexus3``